File: src/main/java/com/example/italianrestaurant/delivery/Delivery.java
package com.example.italianrestaurant.delivery;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.time.LocalDateTime;
import java.util.Objects;

@Entity
@Table(name = "deliveries")
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String address;
    private String city;
    private String postalCode;
    private String floor;
    private String info;
    private LocalDateTime deliveryDate;

    @Enumerated(EnumType.STRING)
    private DeliveryOptions deliveryOptions;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Delivery delivery = (Delivery) o;
        return getId() != null && Objects.equals(getId(), delivery.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: frontend/ItalianRestaurantFront/src/app/models/delivery.ts
export class Delivery{
  address: string
  city: string
  postalCode: string
  floor: string
  info: string
  deliveryOptions: string
  deliveryDate?: Date


  constructor(address: string, city: string, postalCode: string, floor: string, info: string, deliveryOptions: string,deliveryDate? : Date) {
    this.address = address;
    this.city = city;
    this.postalCode = postalCode;
    this.floor = floor;
    this.info = info;
    this.deliveryOptions = deliveryOptions;
    this.deliveryDate = deliveryDate;
  }

}
